xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESettings.cpp

```diff
@@ -23,58 +23,139 @@
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_PROCESSQUALITY);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_ATEMPOTHRESHOLD);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_GUISOUNDMODE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_STEREOUPMIX);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_AC3PASSTHROUGH);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_AC3TRANSCODE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_EAC3PASSTHROUGH);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_DTSPASSTHROUGH);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_TRUEHDPASSTHROUGH);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_DTSHDPASSTHROUGH);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_AUDIODEVICE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_PASSTHROUGHDEVICE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_STREAMNOISE);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_MAINTAINORIGINALVOLUME);
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_DSPADDONSENABLED);
+  CServiceBroker::GetSettings().GetSettingsManager()->RegisterCallback(this, settingSet);
+
+  CServiceBroker::GetSettings().GetSettingsManager()->RegisterSettingOptionsFiller("aequalitylevels",
+                                                                                   SettingOptionsAudioQualityLevelsFiller);
+  CServiceBroker::GetSettings().GetSettingsManager()->RegisterSettingOptionsFiller("audiodevices",
+                                                                                   SettingOptionsAudioDevicesFiller);
+  CServiceBroker::GetSettings().GetSettingsManager()->RegisterSettingOptionsFiller("audiodevicespassthrough",
+                                                                                   SettingOptionsAudioDevicesPassthroughFiller);
+  CServiceBroker::GetSettings().GetSettingsManager()->RegisterSettingOptionsFiller("audiostreamsilence",
+                                                                                   SettingOptionsAudioStreamsilenceFiller);
+}
+
+CActiveAESettings::~CActiveAESettings()
+{
+  CSingleLock lock(m_cs);
+  CServiceBroker::GetSettings().GetSettingsManager()->UnregisterSettingOptionsFiller("aequalitylevels");
+  CServiceBroker::GetSettings().GetSettingsManager()->UnregisterSettingOptionsFiller("audiodevices");
+  CServiceBroker::GetSettings().GetSettingsManager()->UnregisterSettingOptionsFiller("audiodevicespassthrough");
+  CServiceBroker::GetSettings().GetSettingsManager()->UnregisterSettingOptionsFiller("audiostreamsilence");
+  CServiceBroker::GetSettings().GetSettingsManager()->UnregisterCallback(this);
+  m_instance = nullptr;
+}
+
+void CActiveAESettings::OnSettingChanged(std::shared_ptr<const CSetting> setting)
+{
+  CSingleLock lock(m_cs);
+
+  const std::string &settingId = setting->GetId();
+
+  if (settingId == CSettings::SETTING_AUDIOOUTPUT_PASSTHROUGHDEVICE      ||
```

Comment:
would it make sense to simply make the settingsSet created in the ctor a member variable and do a simple m_settingsSet.find(settingId) here instead of duplicating the list of settings? Or would that be a lot slower or have other sideeffects?

Reply:
right, this isn't required at all. if we don't want to notify ae, we just don't register. thx

Reply:
well, that's not what I meant, but now that you mention it, yeah, the callback is only fired for registered events anyways, so the entire if-condition was superfluous :)